docs/source/blockarray.rst

```diff
@@ -0,0 +1,198 @@
+.. _blockarray_class:
+
+BlockArray
+==========
+
+ .. testsetup::
+
+   >>> import scico
+   >>> import scico.numpy as snp
+   >>> from scico.numpy import BlockArray
+   >>> import numpy as np
+   >>> import jax.numpy
+
+The class :class:`.BlockArray` provides a way to combine arrays of
+different shapes into a single object for use with other SCICO classes.
+A :class:`.BlockArray` consists of a list of :obj:`~jax.numpy.DeviceArray` objects,
+which we refer to as blocks. A :class:`.BlockArray` differs from a list in
+that, whenever possible, :class:`.BlockArray` properties and methods
+(including unary and binary operators like +, -, \*, ...) automatically
+map along the blocks, returning another :class:`.BlockArray` or tuple as
+appropriate. For example,
+
+  ::
+
+    >>> x = BlockArray((
+    ...     [[1, 3, 7],
+    ...      [2, 2, 1]],
+    ...     [2, 4, 8]
+    ... ))
+
+    >>> x.shape  # returns tuple
+    ((2, 3), (3,))
+
+    >>> x * 2  # returns BlockArray
```

Comment:
This doctest fails for reasons that are not immediately obvious. Running the test within `ipython` gives the type expected as per stated test result.

Reply:
The tests are run with `JAX_ENABLE_X64=True`, which changes the default dtype of arrays. We can work around it for these docs and I'll make an issue to actually fix it.

Reply:
By work around, do you mean ignore the error and merge, or hold the merge for a fix in a separate PR?